freqtrade/exchange/__init__.py

```diff
@@ -105,132 +94,231 @@ def validate_pairs(pairs: List[str]) -> None:
+            'Could not place buy order due to networking error. Message: {}'.format(e)
+        )
+    except ccxt.BaseError as e:
+        raise OperationalException(e)
 
 
-def sell(pair: str, rate: float, amount: float) -> str:
+def sell(pair: str, rate: float, amount: float) -> Dict:
     if _CONF['dry_run']:
         global _DRY_RUN_OPEN_ORDERS
         order_id = 'dry_run_sell_{}'.format(randint(0, 10**6))
         _DRY_RUN_OPEN_ORDERS[order_id] = {
             'pair': pair,
-            'rate': rate,
+            'price': rate,
             'amount': amount,
-            'type': 'LIMIT_SELL',
+            'type': 'limit',
+            'side': 'sell',
             'remaining': 0.0,
-            'opened': arrow.utcnow().datetime,
-            'closed': arrow.utcnow().datetime,
+            'datetime': arrow.utcnow().isoformat(),
+            'status': 'closed'
         }
-        return order_id
+        return {'id': order_id}
 
-    return _API.sell(pair, rate, amount)
+    try:
+        return _API.create_limit_sell_order(pair, amount, rate)
+    except ccxt.InsufficientFunds as e:
+        raise DependencyException(
+            'Insufficient funds to create limit sell order on market {}.'
+            'Tried to sell amount {} at rate {} (total {}).'
+            'Message: {}'.format(pair, amount, rate, rate*amount, e)
+        )
+    except ccxt.InvalidOrder as e:
+        raise DependencyException(
+            'Could not create limit sell order on market {}.'
+            'Tried to sell amount {} at rate {} (total {}).'
+            'Message: {}'.format(pair, amount, rate, rate*amount, e)
+        )
+    except ccxt.NetworkError as e:
+        raise NetworkException(
+            'Could not place sell order due to networking error. Message: {}'.format(e)
+        )
+    except ccxt.BaseError as e:
+        raise OperationalException(e)
 
 
 def get_balance(currency: str) -> float:
     if _CONF['dry_run']:
         return 999.9
 
-    return _API.fetch_balance()[currency]
+    # ccxt exception is already handled by get_balances
+    balances = get_balances()
+    return balances[currency]['free']
 
 
-def get_balances():
+def get_balances() -> dict:
     if _CONF['dry_run']:
-        return []
+        return {}
+
+    try:
+        balances = _API.fetch_balance()
+        # Remove additional info from ccxt results
+        balances.pop("info", None)
+        balances.pop("free", None)
+        balances.pop("total", None)
+        balances.pop("used", None)
+
+        return balances
+    except ccxt.NetworkError as e:
+        raise NetworkException(
+            'Could not get balance due to networking error. Message: {}'.format(e)
+        )
+    except ccxt.BaseError as e:
+        raise OperationalException(e)
 
-    return _API.fetch_balance()
 
-# @cached(TTLCache(maxsize=100, ttl=30))
 @retrier
 def get_ticker(pair: str, refresh: Optional[bool] = True) -> dict:
-    return _API.fetch_ticker(pair)
+    global _TICKER_CACHE
```

Comment:
I don't think we want to use global ticker-cache. There is e.g. `@cached` from `cachetools` which does it if we need it. My gut feeling is that we should trust the exchanges so that we don't have a cache at all.

Reply:
I agree. Right now it's mostly used in RPC and during selling, which would probably not cause rate limiting.